packages/chart/src/components/BarChart.tsx

```diff
@@ -400,15 +390,7 @@ export default function BarChart({ xScale, yScale, seriesScale, xMax, yMax, getX
                               data-for={`cdc-open-viz-tooltip-${config.runtime.uniqueId}`}
                             ></foreignObject>
                             {orientation === 'horizontal' && !config.isLollipopChart && displayNumbersOnBar && (
-                              <Text
-                                display={displayBar ? 'block' : 'none'}
-                                x={bar.y}
-                                y={config.barHeight / 2 + config.barHeight * bar.index}
-                                fill={labelColor}
-                                dx={doesTextFit ? -5 : 5} // X padding
-                                verticalAnchor='middle'
-                                textAnchor={doesTextFit ? 'end' : 'start'}
-                              >
+                              <Text display={displayBar ? 'block' : 'none'} x={bar.y} y={config.barHeight / 2 + config.barHeight * bar.index} fill={isTextFits ? labelColor : '#000000'} dx={isTextFits ? -5 : 5} verticalAnchor='middle' textAnchor={isTextFits ? 'end' : 'start'}>
```

Comment:
Nit: Consider using `// prettier-ignore` here to keep the line wrapping for readability.

Nit: use the labelColor constant in the fill with your updated logic. That way if it ever needs to be changed we can keep referencing the labelColor variable.